### PR TITLE
protocol_driver_grpc: Fix: Shutdown the client in ShutdownClient instead of in Shutdown server

### PR DESCRIPTION
**Pull request by oserres**

### CODE REVIEW

File: protocol_driver_grpc.cc [hunk 1]
```diff
@@ -96,6 +96,7 @@ void ProtocolDriverGrpc::SetNumPeers(int num_peers) {
 }
 
 ProtocolDriverGrpc::~ProtocolDriverGrpc() {
+  ShutdownClient();
   ShutdownServer();
   grpc_client_stubs_.clear();
```
Comment (danmanj) [inline protocol_driver_grpc.cc:101]: this line of code should be inside shutdown client, and I think we should be calling shutdown server before shutdown client.

File: protocol_driver_grpc_async_callback.cc [hunk 2]
```diff
@@ -104,6 +104,7 @@ void ProtocolDriverGrpcAsyncCallback::SetNumPeers(int num_peers) {
 }
 
 ProtocolDriverGrpcAsyncCallback::~ProtocolDriverGrpcAsyncCallback() {
+  ShutdownClient();
   ShutdownServer();
   grpc_client_stubs_.clear();
```
Comment (danmanj) [inline protocol_driver_grpc_async_callback.cc:109]: this line of code should be inside shutdown client, and I think we should be calling shutdown server before shutdown client.

File: protocol_driver_grpc.cc [hunk 3]
```diff
@@ -198,4 +196,7 @@ void ProtocolDriverGrpc::ShutdownServer() {
   }
 }
 
+void ProtocolDriverGrpc::ShutdownServer() {
+}
```
Comment (danmanj) [inline protocol_driver_grpc.cc:200]: I am doubting that this should be an empty function. Especially now that we are launching threads for the incoming RPC action lisst from inside the protocol driver.

**Review by danmanj — COMMENTED** [review]
I think we should shutting down the server before the client.

File: protocol_driver_grpc.cc (hunk 1 above)
Comment (oserres) [reply protocol_driver_grpc.cc]: Done

File: protocol_driver_grpc_async_callback.cc (hunk 2 above)
Comment (oserres) [reply protocol_driver_grpc_async_callback.cc]: Done

**Review by danmanj — APPROVED** [review]

File: protocol_driver_grpc.cc (hunk 3 above)
Comment (oserres) [reply protocol_driver_grpc.cc:202]: Added code to call Shutdown on the GRPC server

**Pull request merged**